flutter_searchbox/example_cloudSTT/lib/main.dart

```diff
@@ -0,0 +1,157 @@
+import 'package:flutter/material.dart';
+import 'package:searchbase/searchbase.dart';
+import 'package:flutter_searchbox/flutter_searchbox.dart';
+import 'results.dart';
+import 'author_filter.dart';
+import 'audio_recorder.dart';
+import 'mic_overlay.dart';
+
+void main() {
+  runApp(FlutterSearchBoxApp());
+}
+
+class FlutterSearchBoxApp extends StatelessWidget {
+  // Avoid creating searchbase instance in build method
+  // to preserve state on hot reloading
+  final searchbaseInstance = SearchBase(
+      'good-books-ds',
+      'https://appbase-demo-ansible-abxiydt-arc.searchbase.io',
+      'a03a1cb71321:75b6603d-9456-4a5a-af6b-a487b309eb61',
+      appbaseConfig: AppbaseSettings(
+          recordAnalytics: true,
+          // Use unique user id to personalize the recent searches
+          userId: 'jon@appbase.io'));
+
+  FlutterSearchBoxApp({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    // The SearchBaseProvider should wrap your MaterialApp or WidgetsApp. This will
+    // ensure all routes have access to the store.
+    return SearchBaseProvider(
+      // Pass the searchbase instance to the SearchBaseProvider. Any ancestor `SearchWidgetConnector`
+      // widgets will find and use this value as the `SearchController`.
+      searchbase: searchbaseInstance,
+      child: MaterialApp(
+        title: "SearchBox Demo",
+        theme: ThemeData(
+          primarySwatch: Colors.blue,
+          visualDensity: VisualDensity.adaptivePlatformDensity,
+        ),
+        home: HomePage(),
+      ),
+    );
+  }
+}
+
+class HomePage extends StatefulWidget {
+  HomePage();
+
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  OverlayState overlayState;
+  OverlayEntry overlayEntry;
+
+  setOverlay(bool status, String value, BuildContext context) async {
```

Comment:
@anik-ghosh-au7 I'm not able to understand what is going here? Please add comments and use names for variables and methods which are self-exploratory.